### PR TITLE
Chapter1 answers

### PR DESCRIPTION
**Pull request by aiwaverse**
### Solutions for Chapter 1

Hello! After always delaying doing this, decided this hacktoberfest was gonna be it! So this is chapter 1, I already know most of it, but wanted to start from scratch to not miss any information for the next chapters.
Please ignore the stack file, the vscode extension was being weird without it, and adding it solved the issue.
I got a bit confused where I should open the PR, I hope I understood it right.

cc @vrom911 @chshersh

### CODE REVIEW

File: src/Chapter1.hs [hunk 1]
```diff
@@ -490,7 +491,8 @@ Implement a function that returns the last digit of a given number.
   whether it works for you!
 -}
 -- DON'T FORGET TO SPECIFY THE TYPE IN HERE
-lastDigit n = error "lastDigit: Not implemented!"
+lastDigit :: Int -> Int
+lastDigit = (`mod` 10) . abs
```
Comment (vrom911) [inline src/Chapter1.hs:495]: Nice point free style 💅🏼

File: src/Chapter1.hs [hunk 2]
```diff
@@ -632,7 +650,10 @@ Try to introduce variables in this task (either with let-in or where) to avoid
 specifying complex expressions.
 -}
 
-sumLast2 n = error "sumLast2: Not implemented!"
+sumLast2 :: Int -> Int
+sumLast2 n = let digit1 = lastDigit n
```
Comment (vrom911) [inline src/Chapter1.hs:654]: Smart reusage of already introduced function 👍🏼 💡

File: src/Chapter1.hs [hunk 3]
```diff
@@ -568,7 +577,16 @@ True
 >>> isVowel 'x'
 False
 -}
-isVowel c = error "isVowel: not implemented!"
+
+-- I'm aware I can use elem, but I felt this way was on par with the question
+-- And more readable than guards
```
Comment (vrom911) [inline src/Chapter1.hs:582]: Good one! You could also use 
```haskell
isVowel c = c == 'a' 
    || c == 'e'
    || ...
```

But you're right, there are different ways to do that and some are more readable than others 👍🏼

File: src/Chapter1.hs [hunk 4]
```diff
@@ -652,9 +673,9 @@ Implement a function that returns the first digit of a given number.
 You need to use recursion in this task. Feel free to return to it later, if you
 aren't ready for this boss yet!
 -}
-
-firstDigit n = error "firstDigit: Not implemented!"
-
+firstDigit :: Int -> Int
+firstDigit n = go (abs n)
```
Comment (vrom911) [inline src/Chapter1.hs:677]: Fantastic idea on the helper function! 🧠

**Review by vrom911 — APPROVED** [review]
Great Chapter! Ready to be merged! 🚢 

Regarding your question: you did everything correct 👌🏼  If you want to participate in Hacktoberfest though, by rules you need to add `hacktoberfest-accepted` label in the PR and add `hacktoberfest` to the tags of your repo 🙂

**Comment by aiwaverse** [comment]
> 
> 
> Great Chapter! Ready to be merged! 🚢
> 
> Regarding your question: you did everything correct 👌🏼 If you want to participate in Hacktoberfest though, by rules you need to add `hacktoberfest-accepted` label in the PR and add `hacktoberfest` to the tags of your repo 🙂

Thank you very much! I already added the label to the PR, and the repo already had all the tags, thanks for the input.

**Pull request merged**